gridstatus/ieso.py

```diff
@@ -0,0 +1,536 @@
+        """
+        # We have to add special time handling here because the existing utils
+        # don't perfectly handle the 1 hour data files.
+        if isinstance(date, tuple):
+            date, end = date
+
+        today = utils._handle_date("today", tz=self.default_timezone)
+
+        # For today, set date to the start of today and end to now
+        if date == "today":
+            date = today
+            end = pd.Timestamp.now(tz=self.default_timezone)
+        # Latest data returns only a single file for the current hour's data.
+        elif date == "latest":
+            date = pd.Timestamp.now(tz=self.default_timezone)
+            end = None
+
+        # If given a date string or date set date to the earliest interval on the
+        # provided date and end to the latest interval on the date if not specified.
+        elif date != "latest" and (
+            isinstance(date, str) or isinstance(date, datetime.date)
+        ):
+            date = utils._handle_date(date, tz=self.default_timezone).replace(
+                hour=0,
+                minute=0,
+                second=0,
+            )
+            if not end:
+                end = date + pd.Timedelta(days=1)
+
+        if date.date() > today.date():
+            raise NotSupported("Load data is not available for future dates.")
+
+        if date.date() < today.date() - pd.Timedelta(
+            days=MAXIMUM_DAYS_IN_PAST_FOR_LOAD,
+        ):
+            raise NotSupported(
+                f"Load data is not available for dates more than "
+                f"{MAXIMUM_DAYS_IN_PAST_FOR_LOAD} days in the past.",
+            )
+
+        # If end is a datetime, proceed, otherwise handle the end date
+        if end:
+            if not isinstance(end, datetime.datetime):
+                # Set end to the beginning of the next day to ensure we have complete
+                # data from the end date.
+                end = utils._handle_date(
+                    end,
+                    tz=self.default_timezone,
+                ).normalize() + pd.Timedelta(days=1)
+
+            end_with_timezone = (
+                end.tz_localize(self.default_timezone) if end.tzinfo is None else end
+            )
+
+            # Don't request data from after now because it does not exist
+            end = min(pd.Timestamp.now(tz=self.default_timezone), end_with_timezone)
+
+        df = self._retrieve_5_minute_load(date, end, verbose)
+
+        cols_to_keep = [
+            "Interval Start",
+            "Interval End",
+            "Market Total Load",
+            "Ontario Load",
+        ]
+
+        df["Market Total Load"] = df["Market Total Load"].astype(float)
+        df["Ontario Load"] = df["Ontario Load"].astype(float)
+
+        return utils.move_cols_to_front(df, cols_to_keep)[cols_to_keep].reset_index(
+            drop=True,
+        )
+
+    @support_date_range(frequency="HOUR_START")
```

Comment:
There are two other functions that use the "HOUR_START" `frequency`, both in `Ercot`. Using "today" with these functions produces unexpected results:

```py
from gridstatus import IESO, Ercot

ercot = Ercot()

solar = ercot.get_hourly_solar_report("today")
solar[['Interval Start', 'Interval End']].describe().loc[['first', 'last']]

Out[1]: 
                   Interval Start               Interval End
first   2023-12-26 23:00:00-06:00  2023-12-27 00:00:00-06:00
last    2024-01-04 22:00:00-06:00  2024-01-04 23:00:00-06:00


wind = ercot.get_hourly_wind_report("today")
wind[['Interval Start', 'Interval End']].describe().loc[['first', 'last']]

                  Interval Start               Interval End
first  2023-12-26 23:00:00-06:00  2023-12-27 00:00:00-06:00
last   2024-01-04 22:00:00-06:00  2024-01-04 23:00:00-06:00
```

Is this the desired behavior? I want to update the decorator for this frequency, but I need to know how it should behave.

Reply:
supplying today should be same behavior of supplying a date range starting at midnight of current day and ending at midnight the following today. so the following calls should be equiveleent 

`ercot.get_hourly_solar_report("today") == ercot.get_hourly_solar_report(start="12-29-2023", end="12-30-2023")`

Reply:
oh, one thing that might be explaining the behavior your seeing. when a report has a publish time, like the hourly_wind_report, the date are applied to the publish time not the interval start. 

so i would expect the above to return all the hourly reports publish in the range supplied. 

right now, it looks like it just return the last report published before midnight today

Reply:
Those methods are behaving correctly then? If that's the case, I don't think we should update the`support_date_range` decorator only for the `IESO` scraper. That might be a premature generalization.

Reply:
no they arent working correctly, i think this methods is only returning the first interval, which is the same behavior you're seeing in IESO. i suspect if you make it work for IESO, it will fix this issue as well. 

within gridstatus private code base, we always call they methods with intervals, so this today functionality doesn't get used for these reports

Reply:
Got it. I've updated the decorator to correctly handle `frequency="HOUR_START"`.